libs/shared/contentful/src/lib/contentful.service.ts

```diff
@@ -31,11 +33,28 @@ export class ContentfulService {
     offeringConfigId: string,
     interval: SubplatInterval
   ) {
-    const planIds = await this.contentfulManager.getOfferingPlanIds(
-      offeringConfigId
-    );
-    const plan = await this.stripeManager.getPlanByInterval(planIds, interval);
-    if (!plan) throw new PlanNotFoundError();
-    return plan.id;
+    try {
+      const planIds = await this.contentfulManager.getOfferingPlanIds(
+        offeringConfigId
+      );
+      // Temporary supported list of plans
+      // CMS purchase.stripePlanChoices is currently not configured correctly
+      // Unfortunately, currently the CMS is read-only and can't be updated
+      // As a temporary work around provide a list of supported plans
+      const supportedListOfPriceIds =
+        this.contentfulServiceConfig.supportedPlanIds.split(',');
+      const filteredPlanIds = planIds.filter((priceId) =>
+        supportedListOfPriceIds.includes(priceId)
+      );
+      const plan = await this.stripeManager.getPlanByInterval(
+        filteredPlanIds,
+        interval
+      );
+      if (!plan) throw new PlanNotFoundError();
+
+      return plan.id;
+    } catch (error) {
+      throw error;
```

Comment:
Interesting... What does the `try..catch` do here? Since we're just rethrowing the error, do we even need the try vs letting the method just throw on any `await` errors naturally?

Reply:
Oh nothing, this was just for debugging and forgot to remove it.

Reply:
This was copied over into a #16947 and Lisa made the change there. Thank you.